webhook/runtime_webhooks.md

```diff
@@ -0,0 +1,103 @@
+- Add [Runtime](https://www.adobe.io/apis/experienceplatform/console/docs.html#!AdobeDocs/adobeio-console/master/services-enable-runtime.md) to your project, this will create the required auth and runtime namespace for you.
+- Go to `aio-cli` and create and deploy your user-action in your namespace. 
+- Come back to Developer Console and you will see deployed runtime action in your namespace under the `User Defined Actions` section once you refresh.
+- Now set up an event registration using the runtime action you just deployed. You would need to select option 2 (Runtime Action) in your `Configure event registration` page. Choose your runtime action from the dropdown and click `Save configured events`
+- This will create an event registration with an event handler webhook pointing to your runtime action. 
+
+
+## Benefits and Improvements using Runtime Action as Webhook
+
+If your use-case suits having a runtime action that responds within a short time frame, we recommend using the webhook (SYNC based) approach to get the below mentioned benefits.
+
+- [Built In Signature Verification](#built-in-signature-verification)
+- Improved Debugging 
+    - [Debug tracing shows your action response](#debug-tracing-shows-your-action-response)
+    - [Debug tracing shows activation ids for failed invocations](#debug-tracing-shows-activation-ids-for-failed-invocations)
+- [No more duplicate action invocations](#avoiding-duplicate-action-invocations)
+
+
+### Built In Signature Verification
+
+From the security standpoint of webhook invocation, until now there was no way to verify that the event invocation requests to your publicly exposed runtime webhooks are coming from I/O Events and not any malicious user. But, now your runtime webhooks are made secure by being provided with an OOTB signature verification implementation that's working under the hood. So, basically whatever runtime action you use to create an event registration the webhook created due to that will place a signature validator action along with your action as in a [sequence](#https://github.com/AdobeDocs/adobeio-runtime/blob/master/reference/sequences_compositions.md). I/O Events signs the event payload with client's secret signature and pass them in request header to webhook. Whatever requests will come to your webhook will only invoke your business logic runtime action when the validator action successfully verifies that the correct signature has been passed to invoke your webhook.
+
+### Debug Tracing Shows Your Action Response
+
+Debug tracing is a pretty important tool on Developer Console for users who want to be informed whether their webhook invocation is successful or not or what it responds. However, there was a gap in this feature wherein users could not know if their action has succeeded or what the action actually returned, until now. Users could only see an empty response body (as shown below) in the `Debug Tracing` response which was not a good user experience.  
```

Comment:
- I would instead re-use the description in https://www.adobe.io/apis/experienceplatform/events/docs.html#!adobedocs/adobeio-events/master/support/tracing.md and add the link here as well.
- Also, **should we just document how to debug action responses in case something goes wrong, instead of what improved?** Thinking from an end-user perspective, I might not be that interested in what changed, but how to debug my runtime action is important. Documenting the improvements should ideally be part of something like "Release Notes/What's New" which we don't generally do. WDYT?

Reply:
I have the same feedback. this is not a release notes or something that explains what has been fixed or enhanced. most people who come here are first time readers who want to understand what it does and how it works.